### DISCUSSION TITLE
custom provider

### DISCUSSION THREAD

**Discussion by vtbalaji** [nextauthjs/next-auth#2611] [Help]
I am planning to use next-auth,   for one of the trading platform(https://api.icicidirect.com/apiuser/ICICIDirectAPIDOC.htm).
they expect some of the fields part of login, like **AppKey,time_stamp, JSONPostData, Checksum** part of POST.  

I understand it may be deviating from standard.  is there any provision while configuring     Providers I can mention those and pass it.
```js
{
      id: "icici",
      name: "Icici",
      type: "oauth",
      version: "2.0",
      scope: "key",
      params: {
        grant_type: "authorization_code",
      },
      accessTokenUrl: "https://localhost/apiuser/access",
      requestTokenUrl: "http://localhost/apiuser/token",
      authorizationUrl: "https://api.icicidirect.com/apiuser/login",
      profileUrl: "https://www.googleapis.com/oauth2/v1/userinfo?alt=json",
      async profile(profile, tokens) {
        return {
          id: profile.id,
          name: profile.name,
          email: profile.email,
          image: profile.picture,
        };
      },
      clientId: "xyzzy",
      clientSecret: "xys",
    }  
```

**Comment by balazsorban44** [comment]
You can use the `authorizationParams` field: https://next-auth.js.org/configuration/providers#options
